### DISCUSSION TITLE
Unexpected output from Resize

### DISCUSSION THREAD

**Discussion by kayhan-batmanghelich** [Project-MONAI/MONAI#1554] [Q&A] [answered]
Hi,

Thank you for the nice library!

I am trying to make some course material and started playing with the library. I am still not used to it and this question might me being not familiar enough with `monai`. I tried to make a simple `Resize` and I think I am not using the API correctly because output size does not match my expectation. Here is a simple example:

```
sidelength = 64
loader = LoadNifti(image_only=True)
scaler = ScaleIntensity(minv=-1.0,maxv=1.0)
resizer = Resize(spatial_size=(sidelength, sidelength, sidelength), mode=('trilinear'))
tensorer = ToTensor()

inputImg = loader('392272.long.003_S_5150_base/norm.nii.gz')
print(inputImg.shape)

inputImg_scale = scaler(inputImg)
print(inputImg_scale.shape)

inputImg_scale_resize = resizer(inputImg_scale)
print(inputImg_scale_resize.shape)

inputImg_scale_resize_tensor = tensorer(inputImg_scale_resize)
print(inputImg_scale_resize_tensor.shape)
```

This is the output:
```
(256, 256, 256)
(256, 256, 256)
(256, 64, 64, 64)
torch.Size([256, 64, 64, 64])
```

Output supposed to be `(64,64,64)` why is it `(256, 64, 64, 64)`?

**Comment by rijobro** [comment, marked as answer]
Channel normally comes first, so here it assumes you have 256 channels and each image is 256x256 spatially. If you put in an `AddChannel` transform before the resizer, I think that'll solve your problem. 

We also have a `Compose` transform, so once you've sorted out your teething problems, it might be simpler to do this (as you can execute all transforms in one go).

**Reply by kayhan-batmanghelich** [reply]
Thank you @rijobro  That resolves the issue. Perhaps a note in the documentation helps although you have it in the `transfomer`

```
def get_medicalImg_tensor(sidelength):
    xform = Compose([
        LoadNifti(image_only=True),
        AddChannel(),
        ScaleIntensity(minv=-1.0,maxv=1.0),
        Resize((sidelength, sidelength, sidelength), mode=('trilinear')),
        ToTensor(),
    ])
    img = xform('392272.long.003_S_5150_base/norm.nii.gz')
    return img


img = get_medicalImg_tensor(64)
print(img.shape)
```
